algorithms/src/sorting/Kokkos_SortPublicAPI.hpp

```diff
@@ -71,20 +71,34 @@
 
 namespace Kokkos {
 
+// ---------------------------------------------------------------
+// basic overloads
+// ---------------------------------------------------------------
+
+// clang-format off
 template <class ExecutionSpace, class DataType, class... Properties>
-std::enable_if_t<(Kokkos::is_execution_space<ExecutionSpace>::value) &&
-                 (!SpaceAccessibility<
-                     HostSpace, typename Kokkos::View<DataType, Properties...>::
-                                    memory_space>::accessible)>
+std::enable_if_t<
+  (Kokkos::is_execution_space<ExecutionSpace>::value) &&
+  (!SpaceAccessibility<
+     HostSpace, typename Kokkos::View<DataType, Properties...>::memory_space
+    >::accessible)
+  >
 sort(const ExecutionSpace& exec,
-     const Kokkos::View<DataType, Properties...>& view) {
+     const Kokkos::View<DataType, Properties...>& view)
+{
+  // clang-format on
+
+  // despite below we are using BinSort which could work on rank-2 views,
+  // for now view must be rank-1 because the Impl::min_max_functor
+  // used below only works for rank-1 views
```

Comment:
```suggestion
  // Although we are using BinSort below, which could work on rank-2 views,
  // for now view must be rank-1 because the Impl::min_max_functor
  // used below only works for rank-1 views
```